serialize.go

```diff
@@ -43,28 +43,37 @@ func (e *Executor) AsyncExec(f func()) <-chan struct{} {
 	e.init.Do(func() {
 		e.orderCh = make(chan func(), channelSize)
 	})
-	result := make(chan struct{})
-	e.orderCh <- func() {
-		f()
-		close(result)
-	}
-	e.bufferMutex.Lock()
-	e.buffer = append(e.buffer, <-e.orderCh)
-	e.bufferMutex.Unlock()
 	// Start go routine if we don't have one
 	if atomic.AddInt32(&e.count, 1) == 1 {
+		result := make(chan struct{})
 		go func() {
+			f()
+			close(result)
+			if atomic.AddInt32(&e.count, -1) == 0 {
+				return
+			}
 			for {
 				e.bufferMutex.Lock()
-				f := e.buffer[0]
-				e.buffer = e.buffer[1:]
+				buf := e.buffer[0:]
+				e.buffer = e.buffer[len(e.buffer):]
```

Comment:
```suggestion
				e.buffer = nil
```

Reply:
In my observations in the debugger, golang seems smart enough to trim the underlying array for buffer appropriately itself.  I also get better perf leaving that to golang.